package_tarball.md

```diff
@@ -22,3 +22,128 @@ The package tarball contains the following files:
         checksum = sha256(contents)
         final_result = hex_encode(checksum)
 
+
+## Deterministic Archives
+
+The idea of "deterministic" or "reproducible" archives is to empower anyone to
+verify that no flaws have been introduced during the publishing or download
+phase by reproducing byte-for-byte identical Hex packages from a given source
+tree.
+
+Archives (both package tarball and contents.tar.gz) should be compressed in a
+deterministic way. That means, given a package directory, two independent
+packagers can verify that the source directory was the same by comparing
+checksums of the archives.
+
+### Motivation
+
+With reproducible archives, users can declare exact checksums of their
+dependencies, and be sure on what they are depending on. This allows:
+
+1. If package dependencies are declared with a checksum besides name and
+   version, mirrors of hex.pm no longer need to be trusted not to modify the
+   packages. Even more, any distribution channel does not need to be trusted.
+2. If the original tarball is lost ([precedent in npm in early 2016][1]), it
+   can be byte-for-byte reconstructed from the source tree, with assurance it
+   is exactly the same as before.
```

Comment:
this just allows packagers to verify their new package is identical to their old one, it doesn't actually provide any sort of tooling to verify which versions of input artifacts should be used, right?